tasks/discover-granules/index.js

```diff
@@ -192,19 +193,22 @@ const buildGranule = curry(
  *
  * @param {string} granuleId - granuleId to evaluate
  * @param {string} duplicateHandling - collection duplicate handling configuration value
+ * @param {string} params.collectionId - granule collectionId
+ *
  * @returns {*}     - returns granuleId string if no duplicate found, false if
  *                    a duplicate is found.
  * @throws {Error}  - Throws an error on duplicate if
  *                    dupeConfig.duplicateHandling is set to 'error'
  *
  */
-const checkGranuleHasNoDuplicate = async (granuleId, duplicateHandling) => {
+const checkGranuleHasNoDuplicate = async (granuleId, duplicateHandling, collectionId) => {
```

Comment:
added collectionId here as a requirement for the function so that getGranuleResponse would be valid